pkg/controller/sync/dispatch/managed.go

```diff
@@ -68,6 +67,7 @@ func NewManagedDispatcher(clientAccessor clientAccessorFunc, fedResource Federat
 		versionMap:  make(map[string]string),
 	}
 	d.dispatcher = newOperationDispatcher(clientAccessor, d)
+	d.unmanagedDispatcher = newUnmanagedDispatcher(d.dispatcher, d, fedResource.TargetKind(), fedResource.TargetName())
```

Comment:
I think you should be using `NewUnmanagedDispatcher` instead of `newUnmanagedDispatcher`, otherwise the recorder is non-nil and deviates from the main intention of the pr.

Reply:
@shashidharatd This is intentional.  In the managed case (i.e. when using the managed dispatcher), there is a federated resource to log events against even for deletion and unlabeling operations.   That's why the unmanaged dispatcher decides how to log events/errors depending on whether a recorder is available.  In the follow-on PR to refactor deletion handling, the sync controller uses `NewUnmanagedDispatcher` (i.e. without a recorder) to handle deletion and unlabeling when a federated resource is not capable of logging events.

Reply:
Okie, SGTM